src/lib/auth/authconf.rb

```diff
@@ -755,18 +755,99 @@ def aux_apply
+            ) do |one_item|
+                    Yast::Builtins.regexpmatch(one_item, "^[0123456789]+$")
+                end
+            test_rpm_version_l = Yast::Builtins.filter(
+                Yast::Builtins.splitstring(test_rpm_version, "-.+")
+            ) do |one_item|
+                    Yast::Builtins.regexpmatch(one_item, "^[0123456789]+$")
+                end
+
+            Yast::Builtins.y2milestone(
+                "Evaluating installed %1 and test %2 versions",
+                installed_rpm_version_l,
+                test_rpm_version_l
+            )
+
+            installed_version_item = nil
+            test_version_item = nil
+
+            installed_version_is_equal_or_newer = false
+            loop_counter = 0
+            Yast::Builtins.foreach(installed_rpm_version_l) do |i_item|
+                installed_version_item = Yast::Builtins.tointeger(i_item)
+                test_version_item = Yast::Builtins.tointeger(
+                    Yast::Ops.get(test_rpm_version_l, loop_counter, "0")
+                )
+                if test_version_item != installed_version_item
+                    installed_version_is_equal_or_newer = Yast::Ops.greater_or_equal(
+                        installed_version_item,
+                        test_version_item
+                    )
+                    raise Yast::Break
+                end
+                loop_counter = Yast::Ops.add(loop_counter, 1)
+            end
+
+            Yast::Builtins.y2milestone(
+                "%1 >= %2 -> %3",
+                installed_rpm_version,
+                test_rpm_version,
+                installed_version_is_equal_or_newer
+            )
+            installed_version_is_equal_or_newer
+        end
+
         # Create a temporary file holding smb.conf for the specified AD domain.
         # @return [File] a closed file, caller should #unlink after it is no longer used.
         def ad_create_tmp_smb_conf(ad_domain_name, workgroup_name)
+            query_format = "%{VERSION}"
+
+            cmd_installed_rpm_version = Yast::Convert.to_map(
+                Yast::SCR.Execute(
+                    Yast::Path.new(".target.bash_output"),
+                    Yast::Builtins.sformat(
+                        "/bin/rpm -q --queryformat \"%1\" samba",
+                        query_format
+                    )
+                )
+            )
+
+            if Yast::Ops.get_integer(cmd_installed_rpm_version, "exit", -1) != 0
+                Yast::Builtins.y2warning(
+                    "Cannot check the installed samba version: %1",
+                    cmd_installed_rpm_version
+                )
+                Yast::Report.Error(_('Failed to check the installed samba version.'))
+                return nil
+            end
+
+            installed_rpm_version = Yast::Ops.get_string(
+                cmd_installed_rpm_version,
+                "stdout",
+                "undefined-i"
+            )
```

Comment:
And then replace all this code by:
```suggestion
installed_rpm_version = samba_version
if !samba_version
  Yast::Report.Error(_('Failed to check the installed samba version.'))
  return
end
```